pennylane/_qubit_device.py

```diff
@@ -280,7 +280,15 @@ def generate_samples(self):
              array[complex]: array of samples in the shape ``(dev.shots, dev.num_wires)``
         """
         number_of_states = 2 ** self.num_wires
-        rotated_prob = self.probability()
+
+        # Check if device has `analytic_probability` method, otherwise call `probability`
+        # NOTE: try-except statement should be removed when enforcing an `analytic_probabily`
```

Comment:
This looks good and would remind us to carry out this change! Suggesting to change it to `TODO` such that it could be picked up by `pylint` for example.
```suggestion
        # TODO: try-except statement should be removed when enforcing an `analytic_probabily`
```